src/type_info.cpp

```diff
@@ -0,0 +1,68 @@
+///
+/// Copyright 2024 INRIA
+///
+
+#include <typeinfo>
+#include <typeindex>
+
+#include <boost/python.hpp>
+#include <boost/type_index.hpp>
+
+namespace bp = boost::python;
+
+void exposeStdTypeIndex() {
+  typedef std::type_index Self;
+  bp::class_<Self>(
```

Comment:
You can also add a `pretty_name()` method here, using the lambda
```cpp
[] (const Self &value) -> std::string { return boost::core::demangle(value.name()); }
```

Reply:
Perfect. Thanks a lot for the nice suggestion.